keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -50,149 +46,132 @@ status: provisional
-1. If the `BucketClass.bucketIdentifier` is nil, the Sidecar decides the `BucketClass.bucketIdentifier` is a greenfield object store bucket and calls the rpc associated with the `BucketClass.releasePolicy` (*DeleteBucket* or *RevokeBucketAccess*). Otherwise, the `BucketClass.bucketIdentifier` is non-nil, indicating that it is a brownfield object store bucket, and the Sidecar calls the *RevokeBucketAccess()* rpc.
-1. The Sidecar sets `BucketContent.status.phase` to *“Released”.*
-1. The Controller sees `BucketContent` status is “*Released*” and removes `BucketContent`’s `finalizer`.
-1. The `BucketContent` and the dependent `Secret` will be garbage collected.
-1. The Controller removes the `finalizer` from the `Bucket`, allowing it and it’s dependent `secret` to be garbage collected.
 
-### Static Buckets
+### Brownfield: Driverless Bucket Access
 
-> Note: No driver, and thus no sidecar, are present to manage provisioning.
+Clusters with COSI automation deployed but without a provisioner to manage backend bucket operations may still support pseudo-dynamic Bucket access.  Similar to the [Brownfield: Dynamic Bucket Access](#greenfield-dynamic-bucket-creation), the BucketClass will specify a single pre-existing bucket.  
 
-#### Grant Access
+### Manually Managed Buckets
 
-1. An admin defines a `BucketClass` and a `Secret` in a protected namespace, with the `BucketClass.secretRef` field referencing the `Secret`.
-1. The user creates a `Bucket` in their namespace, with reference to a `BucketClass`.
-1. The Controller sees the new `Bucket` and applies a `finalizer` for orchestrated deletions.
-1. The Controller gets the `BucketClass` referenced by the `Bucket`.
-1. The Controller creates a `BucketContent` object with `BucketClassName` set to the name of its `BucketClass`, a `finalizer`, the `secretRef` set to the `BucketClass`’s `secretRef`, and phase set to *“Ready”*.
-1. The Controller detects the `BucketContent` event and sees the *“Ready”* phase. 
-1. The Controller copies the admin defined `secret` to the `Bucket` namespace with name `Bucket.secretName`.  The `secret` is created with an `ownerRef` of the `Bucket`.
-1. The Controller “binds” the `Bucket` and `BucketContent` by setting `bucket.status.bucketContentName` and `bucketContent.bucketRef`, and sets both statuses to *“Bound”*.
-1. The app `pod` ingests `secret` and runs.
+Clusters with zero automation around the COSI API may still benefit from it when deploying COSI aware workload controllers (e.g. a controller managing data analytics jobs).   This case is releveant as it’s likely to be the first step towards adoption for some users.  In this case, admins would define BucketContents themselves and users would create Buckets with a prefilled `bucketContentRef` specifying the BucketContent.  After that, a COSI aware workload controller would be ready to inject connection and credential data from the BucketContent into a workload.
```

Comment:
and there is no need for a Bucket Class? And I assume that the `Bucket`'s status.phase is not filled in to show binding (since user may not be able to do this, right)?